quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow/src/main/java/org/kie/kogito/serverless/workflow/openapi/OpenApiWorkItemHandler.java

```diff
@@ -69,14 +78,29 @@ public void filter(ClientRequestContext requestContext) throws IOException {
                     });
                 }
             }
-        }, Integer.MIN_VALUE).build(clazz);
+        }, Integer.MIN_VALUE).register((ResponseExceptionMapper) response -> new WebApplicationException(fromResponse(response), response.getStatus())).build(clazz);
         try {
             return internalExecute(ref, parameters);
         } catch (WebApplicationException ex) {
             throw new WorkItemExecutionException(Integer.toString(ex.getResponse().getStatus()), ex.getMessage());
         }
     }
 
+    private String fromResponse(Response response) {
+        Object entity = response.getEntity();
+        if (entity instanceof ByteArrayInputStream input) {
+            if (MediaType.APPLICATION_JSON_TYPE.equals(response.getMediaType())) {
+                try {
+                    return ObjectMapperFactory.get().readTree(input).toString();
+                } catch (IOException e) {
+                    logger.warn("Error parsing json error response {}", e.toString());
```

Comment:
why not to log the complete stack trace?
```suggestion
                    logger.warn("Error parsing json error response", e);
```